data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/LogHTTPService.java

```diff
@@ -75,6 +76,32 @@ public HttpResponse doPost(final ServiceRequestContext serviceRequestContext, fi
         return requestProcessDuration.recordCallable(() -> processRequest(aggregatedHttpRequest));
     }
 
+    private void sendChunks(List<String> jsonList, int maxRequestLength) throws Exception {
```

Comment:
We should move this logic into the codec class. It defines how we parse the input data. So it should also decide how to put it back together.

I have some ideas to improve the internal performance. We can do that in a follow-on PR.

Please also add some unit tests as part of the non-draft PR.